quixstreams/state/rocksdb/windowed/transaction.py

```diff
@@ -282,18 +288,67 @@ def expire_windows(
                     # value is [max_timestamp, value] where max_timestamp
                     # is the timestamp of the latest message in the window
                     value[1] = collection
-                collected_expired_windows.append(((start, end), value))
+                collected_expired_windows.append(((start, end), value, key))
             expired_windows = collected_expired_windows
 
         # Delete expired windows from the state
         if delete:
-            for (start, end), _ in expired_windows:
+            for (start, end), _, _ in expired_windows:
                 self.delete_window(start, end, prefix=prefix)
             if collect:
                 self.delete_from_collection(end=start, prefix=prefix)
 
         return expired_windows
 
+    def expire_all_windows(
+        self,
+        max_end_time: int,
```

Comment:
The param is `max_end_time` but the docstring says `max_start_time`. I'm not sure which one is it now but something needs to be corrected.